klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+        klayout_v_list = [int(v) for v in klayout_v_.split(" ")[-1].split(".")]
+
+    logging.info(f"Your Klayout version is: {klayout_v_}")
+
+    if len(klayout_v_list) < 1 or len(klayout_v_list) > 3:
+        logging.error("Was not able to get klayout version properly.")
+        exit(1)
+    elif len(klayout_v_list) == 2:
+        if klayout_v_list[1] < 28:
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+    elif len(klayout_v_list) == 3:
+        if klayout_v_list[1] < 28 :
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+
+
+def check_layout_path(layout_path):
+    """
+    check_layout_type checks if the layout provided is GDS or OAS. Otherwise, kill the process. We only support GDS or OAS now.
+
+    Parameters
+    ----------
+    layout_path : string
+        string that represent the path of the layout.
+
+    Returns
+    -------
+    string
+        string that represent full absolute layout path.
+    """
+
+    if not os.path.isfile(layout_path):
+        logging.error("## GDS file path provided doesn't exist or not a file.")
+        exit(1)
+
+    if ".gds" not in layout_path and ".oas" not in layout_path:
+        logging.error("## Layout is not in GDSII or OASIS format. Please use gds format.")
+        exit(1)
+
+    return os.path.abspath(layout_path)
+
+
+def build_switches_string(sws: dict):
+    """
+    build_switches_string Build swtiches string from dictionary.
+
+    Parameters
+    ----------
+    sws : dict
+        Dictionary that holds the Antenna swithces.
+    """
+    switches_str = ""
+    for k in sws:
+        switches_str += "-rd {}={} ".format(k, sws[k])
+
+    return switches_str
+
+
+def run_check(drc_file: str, drc_name: str, path: str, run_dir: str, sws: dict):
+    """
+    run_antenna_check run DRC check based on DRC file provided.
+
+    Parameters
+    ----------
+    drc_file : str
+        String that has the file full path to run.
+    path : str
+        String that holds the full path of the layout.
+    run_dir : str
+        String that holds the full path of the run location.
+    sws : dict
+        Dictionary that holds all switches that needs to be passed to the antenna checks.
+
+    Returns
+    -------
+    string
+        string that represent the path to the results output database for this run.
+
+    """
+
+    ## Using print because of the multiprocessing
+    logging.info(
+        "Running Global Foundries 180nm MCU {} checks on design {} on cell {}:".format(
+            path, drc_name, sws["topcell"]
+        )
+    )
+
+    layout_base_name = os.path.basename(path).split(".")[0]
```

Comment:
you might consider using https://docs.python.org/3/library/pathlib.html instead, it greatly help with path manipulation.

Reply:
Created a separate issue for that https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/issues/5